linked_list/linked_list.py

```diff
@@ -21,15 +24,52 @@ class LinkedList:
   def __init__(self):
     self.head = None
     self.tail = None
+#we make methods to interact with the node class
 
+#add_to_tail
+#"This involves pointing the next pointer of the the current last node of the linked list
+#to the new data node. So the current last node of the linked list becomes the second last 
+# data node and the new node becomes the last node of the linked list."
   def add_to_tail(self, value):
-    pass
+    if self.head == None and self.tail == None:
+      self.head = Node(value)
+      self.tail = Node(value)
+    else:
+      if self.head.value == None and self.tail.value == None:
+        self.head = Node(value)
+        self.tail = Node(value)
+      else:
+        if self.head.next_node == None:
+          self.head.set_next(value)
+          self.tail = Node(value)
 
+#similar to add_to_tail, but we are assigning a return value that will equal the head and then taking it off
+#making the next value the new head
   def remove_head(self):
-    pass
+    if self.head == None and self.tail == None:
+      return None
 
-  def contains(self, value):
-    pass
+    if self.head.next_node != None:
+      return_value = self.head.value
+      self.head = Node(self.head.next_node)
```

Comment:
When instantiating a node here, a number value should be passed in, not a node type object as it is now.